### PR TITLE
Autocomplete QA updates

### PR DESCRIPTION
**Pull request by ruudniew**

### CODE REVIEW

File: test/integration/channels.js [hunk 1]
```diff
@@ -1789,6 +1789,67 @@ describe('query by $autocomplete operator on channels.name', function () {
...
 		expect(resp[0].cid).to.be.equal(channel.cid);
 	});
+
+	it('empty $autocomplete query should lead to a status 400 error', async () => {
+		let error = false;
+		try {
+			await client.queryChannels({
+				members: [user],
+				name: {
+					$autocomplete: '',
```
Comment (ferhatelmas) [inline test/integration/channels.js:1799]: empty after removal of special characters ?

File: test/integration/query_users.js [hunk 2]
```diff
@@ -278,47 +278,45 @@ describe('Query Users', function () {
...
-				expect(e.response.data.StatusCode).to.equal(400);
-				expect(e.response.data.message).to.equal(
-					'QueryUsers failed with error: "$autocomplete field is empty or contains invalid characters. Please provide a valid string to autocomplete"',
-				);
+		it('$autocomplete query with random characters', async () => {
+			const punctuation = `~\`!@#$%^&*()	_- +=?/>.
+			<,"':;}]{[|\\]`;
+			const azAZ09 =
+				'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
+			const randomString = (length) =>
```
Comment (ferhatelmas) [inline test/integration/query_users.js:286]: good to extract to an utility

File: test/integration/query_users.js [hunk 3]
```diff
@@ -311,14 +298,21 @@ describe('Query Users', function () {
...
-		it('$autocomplete query with special characters, but with valid results', async () => {
-			const result = await client.queryUsers({
-				id: {
-					$autocomplete: `Ru+$#@!`,
-				},
-			});
-			expect(result.users).to.not.be.undefined;
-			expect(result.users.length).to.be.gt(0);
+		it('$autocomplete query with random characters', async () => {
+			for (let i = 0; i < 100; i++) {
```
Comment (ferhatelmas) [inline test/integration/query_users.js:302]: I don't think we need this 100 times

File: test/integration/channels.js [hunk 4]
```diff
@@ -1789,6 +1790,68 @@ describe('query by $autocomplete operator on channels.name', function () {
...
+				},
+			});
+		} catch (e) {
+			error = true;
+			expect(e.response).to.not.be.undefined;
+			expect(e.response.data).to.not.be.undefined;
+			expect(e.response.data.code).to.equal(4);
+			expect(e.response.data.StatusCode).to.equal(400);
+			expect(e.response.data.message).to.equal(
+				'QueryChannels failed with error: "{"code":4,"message":"$autocomplete field is empty or contains invalid characters. Please provide a valid string to autocomplete","StatusCode":400,"duration":"","more_info":""}"',
```
Comment (ferhatelmas) [inline test/integration/channels.js:1810]: no need for this since code and status are checked

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Comment by ferhatelmas** [comment]
@thesyncim fine for me but since I made some change, PGAL

**Review by thesyncim — APPROVED** [review]

**Pull request merged**